src/library.js

```diff
@@ -3612,6 +3614,20 @@ addToLibrary({
   $getNativeTypeSize__deps: ['$POINTER_SIZE'],
   $getNativeTypeSize: {{{ getNativeTypeSize }}},
 
+#if RELOCATABLE
+  // In RELOCATABLE mode we create the table in JS.
+  $wasmTable: `=new WebAssembly.Table({
+  'initial': {{{ INITIAL_TABLE }}},
+#if !ALLOW_TABLE_GROWTH
+  'maximum': {{{ INITIAL_TABLE }}},
+#endif
+  'element': 'anyfunc'
+});
+`,
+#else
+  $wasmTable: undefined,
```

Comment:
Does this not confuse deps management? I guess we can see the key exists in the map even if the value is undefined. This is slightly magical behaviour, though. Perhaps this could be set to `null`, which would also avoid that magic in `jsifier.js`?

(or if we set the string `undefined` here then at runtime it would be `undefined`, and closure could even optimize it out I think?)

Reply:
We want to the result in the file to be `var foo;`  and not `var foo = undefined` or `var foo = null` which would be needlessly wasteful.  So I think we need a special case for this in jsifier.

Given that the initial value of the variable will be `undefined` it seems to be that using `undefined` here make more sense than `null`.. but I've never really used `null` in JS.

Reply:
WDYT?

Reply:
It's probably fine either way. It's just the extra magic in `jsifier.js` that I'd like to avoid, which `$wasmTable: 'undefined',` (as a string) would do for example. Maybe worth seeing if closure optimizes that assignment away (which it could)?

Reply:
Maybe closure could optimize it away, but I also like producing compact output in the non-closure.   

Just declaring a variable without assigning anything to it seems like a fairly useful thing to do be able to do.

Reply:
Fair enough, lgtm as is.